Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Utilities.ps1

```diff
@@ -232,24 +232,190 @@ function Get-ServiceFabricApplicationAction
     return Get-ServiceFabricApplication @getApplicationParams
 }
 
+function Get-ServiceFabricApplicationUpgradeAction
+{
+    Param (
+        [string]
+        $ApplicationName
+    )
+
+    $global:operationId = $SF_Operations.GetApplicationUpgradeStatus
+    return Get-ServiceFabricApplicationUpgrade -ApplicationName $ApplicationName
+}
+
+function Copy-ServiceFabricApplicationPackageAction
+{
+    Param (
+        [hashtable]
+        $CopyParameters
+    )
+
+    $global:operationId = $SF_Operations.CopyApplicationPackage
+    $copyAction = { Copy-ServiceFabricApplicationPackage @CopyParameters }
+
+    Invoke-ActionWithDefaultRetries -Action $copyAction `
+        -RetryMessage (Get-VstsLocString -Key SFSDK_RetryingCopyApplicationPackage)
+}
+
+function Register-ServiceFabricApplicationTypeAction
+{
+    Param (
+        [hashtable]
+        $RegisterParameters,
+
+        [string]
+        $ApplicationTypeName,
+
+        [string]
+        $ApplicationTypeVersion
+    )
+
+    $global:operationId = $SF_Operations.RegisterApplicationType
+    $registerAction = { Register-ServiceFabricApplicationType @RegisterParameters }
+
+    $exceptionRetryEvaluator = {
+        param($ex)
+        $appType = Get-ServiceFabricApplicationTypeAction -ApplicationTypeName $ApplicationTypeName -ApplicationTypeVersion $ApplicationTypeVersion
```

Comment:
This is Incorrect. You are entering into infinite loop. There will be no bound of MaxRetries.

Reply:
Get-ServiceFabricApplicationTypeAction has a max tries of 3. Or are you pointing to something different? Can you explain how it will go into infinit?

Reply:
Oops my mistake. I thought function name is also Register-ServiceFabricApplicationTypeAction.
This is correct!